src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -378,14 +385,20 @@ public async Task<IActionResult> SetDeploymentTarget(string sessionId, [FromBody
                     return NotFound($"Recommendation {input.NewDeploymentRecipeId} used in existing deployment {existingDeployment.RecipeId} not found.");
                 }
 
-                var existingCloudApplicationMetadata = await templateMetadataReader.LoadCloudApplicationMetadata(input.ExistingDeploymentName);
-                state.SelectedRecommendation = state.SelectedRecommendation.ApplyPreviousSettings(existingCloudApplicationMetadata.Settings);
+                IDictionary<string, object> previousSettings;
+                if (existingDeployment.ResourceType == CloudApplicationResourceType.CloudFormationStack)
+                    previousSettings = (await templateMetadataReader.LoadCloudApplicationMetadata(existingDeployment.Name)).Settings;
+                else
+                    previousSettings = await deployedApplicationQueryer.GetPreviousSettings(existingDeployment);
+
+                state.SelectedRecommendation = state.SelectedRecommendation.ApplyPreviousSettings(previousSettings);
```

Comment:
Minor: Consider moving the `state.SelectedRecommendation` assignment directly above where you make the `AddReplacementToken` call on it, to group relevant code together.